frontend/src/components/ShootWorkers/WorkerGroup.vue

```diff
@@ -66,81 +67,122 @@ export default {
-          value: join(this.workerGroup.zones, ', ')
+          title: zones.length > 1 ? 'Zones' : 'Zone',
+          value: join(zones, ', ')
         })
       }
+
       return description
     }
+  },
+  methods: {
+    getMachineTpeDescription () {
+      const { machine } = this.workerGroup
+      const item = {
+        title: 'Machine Type',
+        value: machine.type
+      }
+
+      const machineMetadata = find(this.machineTypes, ['name', machine.type])
+      if (machineMetadata) {
+        item.description = `(CPU: ${machineMetadata.cpu} | GPU: ${machineMetadata.gpu} | Memory: ${machineMetadata.memory})`
+      }
+
+      return item
+    },
+    getVolumeTypeDescription () {
+      // workers with volume type (e.g. aws)
+      const { volume } = this.workerGroup
+      if (volume && volume.type) {
+        const item = {
+          title: 'Volume Type',
+          value: volume.type
+        }
+
+        const volumeMetadata = find(this.volumeTypes, ['name', volume.type])
+        if (volumeMetadata) {
+          item.description = `(Class: ${volumeMetadata.class})`
+        }
+
+        return item
+      }
+
+      // workers with storage in machine type metadata (e.g. openstack)
+      const { machine } = this.workerGroup
+      const machineMetadata = find(this.machineTypes, ['name', machine.type])
+      const storageType = get(machineMetadata, 'storage.type')
+      if (storageType) {
+        return {
+          title: 'Volume Type',
+          value: storageType,
+          description: `(Class: ${machineMetadata.storage.class})`
+        }
+      }
+    },
+    getVolumeSizeDescription () {
+      // all infras support volume sizes, but for some they are optional
+      const { volume } = this.workerGroup
+      const volumeSize = get(volume, 'size')
+      if (volume && volume.size) {
+        return {
+          title: 'Volume Size',
+          value: `${volumeSize}`
+        }
+      }
+
+      // if no size is defined on the worker itself, check if machine storage defines a default size
+      const { machine } = this.workerGroup
+      const machineMetadata = find(this.machineTypes, ['name', machine.type])
+      const storageSize = get(machineMetadata, 'storage.size')
+      if (storageSize) {
+        return {
+          title: 'Volume Size',
+          value: `${storageSize}`
+        }
+      }
+    },
+    getMachineImageDescription () {
+      const { machine: { image: { name, version } } } = this.workerGroup
+      const item = {
+        title: 'Machine Image',
+        value: `${name} | Version: ${version}`
+      }
+
+      const machineImageMetadata = find(this.machineImages, { name, version })
+      if (!machineImageMetadata) {
+        item.description = '(Image is expired)'
+      } else if (machineImageMetadata.expirationDate) {
+        item.description = `(Expires: ${machineImageMetadata.expirationDateString})`
+      }
+
+      return item
+    }
```

Comment:
```suggestion
    machineType () {
      const type = get(this.workerGroup, 'machine.type')
      return find(this.machineTypes, ['name', type])
    },
    volumeType () {
      const type = get(this.workerGroup, 'volume.type')
      return find(this.volumeTypes, ['name', type])
    },
    machineImage () {
      const { name, version } = get(this.workerGroup, 'machine.image', {})
      return find(this.machineImages, { name, version })
    },
    machineTypeDescription () {
      const machine = get(this.workerGroup, 'machine', {})
      const item = {
        title: 'Machine Type',
        value: machine.type
      }

      const machineType = this.machineType
      if (machineType) {
        const { cpu = '', gpu = '', memory = '' } = machineType
        item.description = `(CPU: ${cpu} | GPU: ${gpu} | Memory: ${memory})`
      }

      return item
    },
    volumeTypeDescription () {
      // workers with volume type (e.g. aws)
      const volume = get(this.workerGroup, 'volume', {})
      if (volume.type) {
        const item = {
          title: 'Volume Type',
          value: volume.type
        }

        const volumeType = this.volumeType
        if (volumeType) {
          item.description = `(Class: ${volumeType.class})`
        }

        return item
      }

      // workers with storage in machine type metadata (e.g. openstack)
      const storage = get(this.machineType, 'storage', {})
      if (storage.type) {
        return {
          title: 'Volume Type',
          value: storage.type,
          description: `(Class: ${storage.class})`
        }
      }
    },
    volumeSizeDescription () {
      // all infras support volume sizes, but for some they are optional
      const volume = get(this.workerGroup, 'volume', {})
      if (volume.size) {
        return {
          title: 'Volume Size',
          value: `${volume.size}`
        }
      }

      // if no size is defined on the worker itself, check if machine storage defines a default size
      const storage = get(this.machineType, 'storage', {})
      if (storage.size) {
        return {
          title: 'Volume Size',
          value: `${storage.size}`
        }
      }
    },
    machineImageDescription () {
      const { name, version }  = get(this.workerGroup, 'machine.image', {})
      const item = {
        title: 'Machine Image',
        value: `${name} | Version: ${version}`
      }

      const machineImage = this.machineImage
      if (!machineImage) {
        item.description = '(Image is expired)'
      } else if (machineImage.expirationDate) {
        item.description = `(Expires: ${machineImage.expirationDateString})`
      }

      return item
    }
```

Reply:
If there is a machineImage but it has no expirationDate we do not show a description. Should it be (Image is up to date)?